test/src/unit-capi-consolidation.cc

```diff
@@ -7590,13 +7590,27 @@ TEST_CASE_METHOD(
 TEST_CASE_METHOD(
     ConsolidationFx,
     "C API: Test consolidation, sparse string, no progress",
-    "[capi][consolidation][sparse][string][no-progress][non-rest]") {
+    "[!shouldfail][capi][consolidation][sparse][string][no-progress][non-"
+    "rest]") {
   remove_sparse_string_array();
   create_sparse_string_array();
 
   write_sparse_string_full();
   write_sparse_string_unordered();
-  consolidate_sparse_string(1, true);
+
+  uint64_t string_size = 1;
+
+  SECTION("too small") {
+    string_size = 1;
+  }
+
+  // Consolidation currently fails to respect the memory budget.
+  // Once this is resolved, the test can be unmarked as `shouldfail`.
+  SECTION("too large") {
+    string_size = 60 * 10485760;  // 6 GB (over default memory budget of 5GB)
+  }
+
+  consolidate_sparse_string(string_size, true);
```

Comment:
Does this write only the one input cell?
If so then the write queue is always empty for this test.
We should have additional tests which contain huge cells which occur when the write queue has data in it.  e.g. the buffer is 4G, the writer is processing 1G, there are two buffers queued at 1G each, and the next cell is 2G.
We must have coverage for waiting for there to be enough space to read more data.